sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentitySyncClient.java

```diff
@@ -0,0 +1,357 @@
+            ? new SynchronousAccessor<>(() -> parseClientAssertion(), Duration.ofMinutes(5))
+            : new SynchronousAccessor<>(() -> parseClientAssertion(), clientAssertionTimeout);
+    }
+
+    private String parseClientAssertion() {
+        if (clientAssertionFilePath != null) {
+            byte[] encoded = new byte[0];
+            try {
+                encoded = Files.readAllBytes(Paths.get(clientAssertionFilePath));
+            } catch (IOException e) {
+                throw LOGGER.logExceptionAsError(new RuntimeException(e));
+            }
+            return new String(encoded, StandardCharsets.UTF_8);
+        } else {
+            throw LOGGER.logExceptionAsError(new IllegalStateException(
+                "Client Assertion File Path is not provided."
+                    + " It should be provided to authenticate with client assertion."
+            ));
+        }
+    }
+
+    /**
+     * Asynchronously acquire a token from Active Directory with a client secret.
+     *
+     * @param request the details of the token request
+     * @return a Publisher that emits an AccessToken
+     */
+    public AccessToken authenticateWithConfidentialClient(TokenRequestContext request) {
+        ConfidentialClientApplication confidentialClient =  confidentialClientApplicationAccessor.getValue();
+        ClientCredentialParameters.ClientCredentialParametersBuilder builder =
+            ClientCredentialParameters.builder(new HashSet<>(request.getScopes()))
+                .tenant(IdentityUtil
+                    .resolveTenantId(tenantId, request, options));
+        if (clientAssertionSupplier != null) {
+            builder.clientCredential(ClientCredentialFactory
+                .createFromClientAssertion(clientAssertionSupplier.get()));
+        }
+        try {
+            return new MsalToken(confidentialClient.acquireToken(builder.build()).get());
+        } catch (InterruptedException | ExecutionException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+        }
+    }
+
+    public AccessToken authenticateWithManagedIdentityConfidentialClient(TokenRequestContext request) {
+        ConfidentialClientApplication confidentialClient =  managedIdentityConfidentialClientApplicationAccessor.getValue();
+        ClientCredentialParameters.ClientCredentialParametersBuilder builder =
+            ClientCredentialParameters.builder(new HashSet<>(request.getScopes()))
+                .tenant(IdentityUtil
+                    .resolveTenantId(tenantId, request, options));
+        try {
+            return new MsalToken(confidentialClient.acquireToken(builder.build()).get());
+        } catch (Exception e) {
+            throw new CredentialUnavailableException("Managed Identity authentication is not available.", e);
+        }
+    }
+
+
+    public AccessToken authenticateWithConfidentialClientCacheSync(TokenRequestContext request) {
```

Comment:
Why is this the only method in this class that ends in `Sync`? Should they all be? Or should this one have it removed?